content/en/docs/tutorials/configuration/updating-configuration-via-a-configmap.md

```diff
@@ -0,0 +1,514 @@
+{{< glossary_tooltip text="selector" term_id="selector" >}}):
+```shell
+kubectl get pods --selector=app.kubernetes.io/name=configmap-env-var
+```
+
+You should see an output similar to:
+```
+NAME                                 READY   STATUS    RESTARTS   AGE
+configmap-env-var-59cfc64f7d-74d7z   1/1     Running   0          46s
+configmap-env-var-59cfc64f7d-c4wmj   1/1     Running   0          46s
+configmap-env-var-59cfc64f7d-dpr98   1/1     Running   0          46s
+```
+
+The key-value pair in the ConfigMap is configured as an environment variable in the container of the Pod.
+Check this by viewing the logs of the Pod.
+```shell
+kubectl logs deployment/configmap-env-var
+```
+
+You should see an output similar to:
+```
+Found 3 pods, using pod/configmap-env-var-7c994f7769-l74nq
+Thu Jan  4 16:07:06 UTC 2024 The basket is full of apples
+Thu Jan  4 16:07:16 UTC 2024 The basket is full of apples
+Thu Jan  4 16:07:26 UTC 2024 The basket is full of apples
+```
+
+Edit the ConfigMap:
+```shell
+kubectl edit configmap fruits
+```
+
+In the editor that appears, change the value of key `fruits` from `apples` to `mangoes`. Save your changes.
+The kubectl tool updates the ConfigMap accordingly (if you see an error, try again).
+
+Here's an example of how that manifest could look after you edit it:
+```yaml
+apiVersion: v1
+data:
+  fruits: mangoes
+kind: ConfigMap
+# You can leave the existing metadata as they are.
+# The values you'll see won't exactly match these.
+metadata:
+  creationTimestamp: "2024-01-04T16:04:19Z"
+  name: fruits
+  namespace: default
+  resourceVersion: "1749472"
+
+```
+
+You should see the following output:
+```
+configmap/fruits edited
+```
+
+Tail the logs of the Deployment and observe the output for few seconds:
+```shell
+kubectl logs deployment/configmap-env-var
+```
+
+Notice that the output remains **unchanged**, even though you edited the ConfigMap:
+```
+Thu Jan  4 16:12:56 UTC 2024 The basket is full of apples
+Thu Jan  4 16:13:06 UTC 2024 The basket is full of apples
+Thu Jan  4 16:13:16 UTC 2024 The basket is full of apples
+Thu Jan  4 16:13:26 UTC 2024 The basket is full of apples
+
+```
+
+{{< note >}}
+Although the value of the key inside the ConfigMap has changed, the environment variable in the Pod still shows the earlier value. This is because environment variables for a process running inside a Pod are **not** updated when the source data changes; if you wanted to force an update, you would need to have Kubernetes replace your existing Pods. The new Pods would then run with the updated information.
+{{< /note >}}
+
+You can trigger that replacement. Perform a rollout for the Deployment, using
+[`kubectl rollout`](/docs/reference/kubectl/generated/kubectl_rollout/):
+```shell
+# Trigger the rollout
+kubectl rollout restart deployment configmap-env-var
+
+# Wait for the rollout to complete
+kubectl rollout status deployment configmap-env-var --watch=true
+```
+
```

Comment:
There's a problem; `kubectl rollout` does not actually wait for the old Pods to disappear. The old Pods will still be around, for a short time, albeit `Terminating`.

Reply:
This works, but what a lot to type:
```shell
kubectl rollout restart deployment configmap-env-var && \
  kubectl rollout status deployment configmap-env-var --watch=true && \
  while kubectl get pods --selector=app.kubernetes.io/name=configmap-env-var \
  --template '{{range .items}}{{ if .metadata.deletionTimestamp }}pods/{{.metadata.name}}{{"\n"}}{{end}}{{end}}' |
  grep -q . ; do sleep 1.5; done && echo Rollout complete 1>&2
```

(I wonder if `kubectl rollout` should support waiting for the selected Pods to actually terminate, or if the Deployment and ReplicaSet controllers should somehow track the Pods that they once owned right through to deletion)

Reply:
Thanks for the alternate command, but I agree it is too much to type for this tutorial and we would not want to confuse the readers.

I will handle it by modifying the text to clearly imply that the readers should wait till the older Pods terminate completely before proceeding with the tutorial.